src/task_impl/atomic_task.rs

```diff
@@ -7,7 +7,25 @@ use core::cell::UnsafeCell;
 use core::sync::atomic::AtomicUsize;
 use core::sync::atomic::Ordering::{Acquire, Release};
 
-/// A coordinated `Task` handle enabling concurrent operations on a task.
+/// A synchronization primitive for task notification.
+///
+/// `AtomicTask` will coordinate concurrent notifications with the consumer
+/// potentially "updating" the underlying task to notify. This is useful in
+/// scenarios where a computation completes in another thread and wants to
+/// notify the consumer, but the consumer is in the process of being migrated to
+/// a new logical task.
+///
+/// Roughly, consumers should call `park` before checking the result of a
```

Comment:
Hm ok reading over the park/unpark implementation again it turns out it's significantly different in this aspect than I was expecting. I initially expected this to behave like `std::thread`'s park/unpark where if you call `unpark` (or `notify` in this case) it "buffers a notification" if there's no blocking task. I think that would mean that you're not *required* to call `park` before you do any work. Do you know if there's a reason to *not* implement that?